pkg/pagerduty/pagerduty.go

```diff
@@ -126,6 +126,38 @@ func (p PagerDuty) AddNote(incidentID string, noteContent string) error {
 	return nil
 }
 
+// GetAlerts will retrieve the alerts for a specific incident
+func (p PagerDuty) GetAlerts(incidentID string) ([]Alert, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), pagerDutyTimeout)
+	defer cancel()
+
+	o := sdk.ListIncidentAlertsOptions{}
+
+	alerts, err := p.c.ListIncidentAlertsWithContext(ctx, incidentID, o)
+	if err != nil {
+		sdkErr := sdk.APIError{}
+		if errors.As(err, &sdkErr) {
+			err := commonErrorHandling(err, sdkErr)
+			if err != nil {
+				return nil, err
+			}
+			if sdkErr.StatusCode == http.StatusNotFound {
+				// this case can happen if the incidentID is not a valid incident (like a number prepended with zeroes)
+				return nil, IncidentNotFoundErr{Err: err}
+			}
+		}
+		return nil, UnknownGetAlertsError{Err: err}
+	}
+
+	res := []Alert{}
+	for i := range alerts.Alerts {
```

Comment:
whats about? :
```
for i, alert := range alerts.Alerts {
	localAlert := toLocalAlert(alert)
	res = append(res, localAlert)
}
```

Reply:
I think there was something with `gosec` that didn't want to use this format
@NautiluX might remember the card

Reply:
yes, I think we had the issue here:
https://github.com/openshift/route-monitor-operator/commit/25e839a441cccfb1bdcad1e176c9713000096b84
However, in this case is a bit different IMO.
We are creating a new struct in toLocalAlert(alert) and appending this.
So we are not using the memory address of the iterator element.

Reply:
let's leave it as is? maybe add a note in a different card regarding this

Reply:
we would have been able to save one line by utilizing the elegant syntax :) But yes its nitpicking

Reply:
checked and gosec doesn't complain. so I'll leave the card but use your syntax